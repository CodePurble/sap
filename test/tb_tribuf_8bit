#! /usr/bin/vvp
:ivl_version "11.0 (stable)" "(v11_0)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/usr/lib/ivl/system.vpi";
:vpi_module "/usr/lib/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib/ivl/v2005_math.vpi";
:vpi_module "/usr/lib/ivl/va_math.vpi";
S_0x556013e7e7c0 .scope module, "tb_tribuf_8bit" "tb_tribuf_8bit" 2 1;
 .timescale 0 0;
v0x556013e9f4c0_0 .var "enable", 0 0;
v0x556013e9f560_0 .var "in", 7 0;
v0x556013e9f600_0 .net "out", 7 0, L_0x556013e9f6d0;  1 drivers
S_0x556013e7e950 .scope module, "uut" "tribuf_8bit" 2 6, 3 1 0, S_0x556013e7e7c0;
 .timescale 0 0;
    .port_info 0 /INPUT 8 "in";
    .port_info 1 /OUTPUT 8 "out";
    .port_info 2 /INPUT 1 "enable";
o0x7f778d720018 .functor BUFZ 8, C4<zzzzzzzz>; HiZ drive
; Elide local net with no drivers, v0x556013e580e0_0 name=_ivl_0
v0x556013e9f1e0_0 .net "enable", 0 0, v0x556013e9f4c0_0;  1 drivers
v0x556013e9f2a0_0 .net "in", 7 0, v0x556013e9f560_0;  1 drivers
v0x556013e9f360_0 .net "out", 7 0, L_0x556013e9f6d0;  alias, 1 drivers
L_0x556013e9f6d0 .functor MUXZ 8, v0x556013e9f560_0, o0x7f778d720018, v0x556013e9f4c0_0, C4<>;
    .scope S_0x556013e7e7c0;
T_0 ;
    %vpi_call 2 10 "$dumpfile", "../simulation/tb_tribuf_8bit.vcd" {0 0 0};
    %vpi_call 2 11 "$dumpvars", 32'sb00000000000000000000000000000000, S_0x556013e7e7c0 {0 0 0};
    %pushi/vec4 204, 0, 8;
    %store/vec4 v0x556013e9f560_0, 0, 8;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x556013e9f4c0_0, 0, 1;
    %vpi_call 2 14 "$monitor", "out:%b", v0x556013e9f600_0 {0 0 0};
    %end;
    .thread T_0;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "/home/ramprakash/repos/sap/test/tb_tribuf_8bit.v";
    "../hdl/tribuf_8bit.v";
