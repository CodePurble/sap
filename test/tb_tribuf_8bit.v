module tb_tribuf_8bit;
	reg [7:0] in;
	reg low_enable;
	wire [7:0] out;

	tribuf_8bit uut(.in(in), .out(out), .low_enable(low_enable));

	initial
	begin
		$dumpfile("../simulation/tb_tribuf_8bit.vcd");
		$dumpvars(0, tb_tribuf_8bit);
		in = 8'b11001100;
		low_enable = 1;
		$monitor("out:%b", out);
	end
endmodule
