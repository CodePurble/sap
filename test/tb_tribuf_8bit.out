#! /usr/bin/vvp
:ivl_version "11.0 (stable)" "(v11_0)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/usr/lib/ivl/system.vpi";
:vpi_module "/usr/lib/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib/ivl/v2005_math.vpi";
:vpi_module "/usr/lib/ivl/va_math.vpi";
S_0x56097ccf67c0 .scope module, "tb_tribuf_8bit" "tb_tribuf_8bit" 2 1;
 .timescale 0 0;
v0x56097cd17490_0 .var "enable", 0 0;
v0x56097cd17530_0 .var "in", 7 0;
v0x56097cd175d0_0 .net "out", 7 0, L_0x56097cd176a0;  1 drivers
S_0x56097ccf6950 .scope module, "uut" "tribuf_8bit" 2 6, 3 2 0, S_0x56097ccf67c0;
 .timescale 0 0;
    .port_info 0 /INPUT 8 "in";
    .port_info 1 /OUTPUT 8 "out";
    .port_info 2 /INPUT 1 "enable";
o0x7fa5579e2018 .functor BUFZ 8, C4<zzzzzzzz>; HiZ drive
; Elide local net with no drivers, v0x56097ccd00e0_0 name=_ivl_0
v0x56097cd171b0_0 .net "enable", 0 0, v0x56097cd17490_0;  1 drivers
v0x56097cd17270_0 .net "in", 7 0, v0x56097cd17530_0;  1 drivers
v0x56097cd17330_0 .net "out", 7 0, L_0x56097cd176a0;  alias, 1 drivers
L_0x56097cd176a0 .functor MUXZ 8, v0x56097cd17530_0, o0x7fa5579e2018, v0x56097cd17490_0, C4<>;
    .scope S_0x56097ccf67c0;
T_0 ;
    %vpi_call 2 10 "$dumpfile", "../simulation/tb_tribuf_8bit.vcd" {0 0 0};
    %vpi_call 2 11 "$dumpvars", 32'sb00000000000000000000000000000000, S_0x56097ccf67c0 {0 0 0};
    %pushi/vec4 204, 0, 8;
    %store/vec4 v0x56097cd17530_0, 0, 8;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x56097cd17490_0, 0, 1;
    %vpi_call 2 14 "$monitor", "out:%b", v0x56097cd175d0_0 {0 0 0};
    %end;
    .thread T_0;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "tb_tribuf_8bit.v";
    "../hdl/tribuf_8bit.v";
